src/test/javascript/spec/component/programming-exercise/feedback-analysis/modals/feedback-filter-modal.component.spec.ts

```
@@ -0,0 +1,98 @@
+        expect(component.filterForm.value.tasks).toEqual(['test-task']);
+    });
+
+    it('should remove the value from form control when checkbox is unchecked', () => {
+        component.filterForm.get('tasks')?.setValue(['test-task', 'task-2']);
+        const event = { target: { checked: false, value: 'test-task' } } as unknown as Event;
+        component.onCheckboxChange(event, 'tasks');
+        expect(component.filterForm.value.tasks).toEqual(['task-2']);
+    });
+
+    it('should set occurrence correctly using the setter', () => {
+        const setValueSpy = jest.spyOn(component.filterForm.get('occurrence')!, 'setValue');
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)